raws-feed-processor/src/main/java/edu/mit/ll/nics/processor/raws/RAWSFeedProcessor.java

```diff
@@ -0,0 +1,125 @@
+        if(rawsFeatures.size() == 0) {
+            logger.info("Zero RAWS features to process, exiting current process");
+            return;
+        }
+        logger.debug(String.format("Processing %d RAWS features", rawsFeatures.size()));
+        DefaultTransaction transaction = null;
+        int successfulFeatureUpdates = 0, failedFeatureUpdates = 0;
+
+        try {
+            DataStore datastore = dataStoreManager.getInstance();
+            SimpleFeatureSource featureSource = datastore.getFeatureSource(rawsFeatureSource);
+            if (!(featureSource instanceof SimpleFeatureStore)) {
+                logger.error(String.format("Write denied on feature source : %s", featureSource));
+                return;
+            }
+
+            SimpleFeatureStore featureStore = (SimpleFeatureStore) featureSource;
+            transaction = new DefaultTransaction("raws-feed-tx");
+            featureStore.setTransaction(transaction);
+            SimpleFeatureType rawsFeatureType = featureStore.getSchema();
+            SimpleFeatureBuilder featureBuilder = new SimpleFeatureBuilder(rawsFeatureType);
+            for(RAWSFeature rawsFeature : rawsFeatures) {
+                if(this.updateFeature(rawsFeature, featureBuilder, featureStore, transaction, rawsFeatureType))
+                    successfulFeatureUpdates++;
+                else
+                    failedFeatureUpdates++;
+                if((successfulFeatureUpdates+failedFeatureUpdates)%100 == 0)
+                    logger.debug(String.format("So far processed %d features", (successfulFeatureUpdates+failedFeatureUpdates)));
+            }
+        } catch(Exception e) {
+            logger.error(String.format("Failed to process RAWS features successfully, ran for %d ms before failing", stopwatch.elapsed(TimeUnit.MILLISECONDS)), e);
+            if(transaction != null)
+                transaction.rollback();
+        } finally {
+            if(transaction != null)
+                transaction.close();
+        }
+        logger.info(String.format("Successfully completed processing %d RAWS Features & Failed to process %d RAWS Features in %d ms", successfulFeatureUpdates, failedFeatureUpdates, stopwatch.elapsed(TimeUnit.MILLISECONDS)));
+    }
+
+    public boolean updateFeature(RAWSFeature rawsFeature, SimpleFeatureBuilder featureBuilder, SimpleFeatureStore featureStore, DefaultTransaction transaction, SimpleFeatureType rawsFeatureType) throws Exception {
+        try {
+            Filter filter = CQL.toFilter("station_id = '" + rawsFeature.getRawsObservations().getStationId() + "'");
+            featureStore.removeFeatures(filter);
+            if(rawsFeature.getRawsObservations().getStatus().equalsIgnoreCase("ACTIVE")) { //add feature only if station status is active
+                addFeature(rawsFeature, featureBuilder, featureStore, rawsFeatureType);
+            }
+            transaction.commit();
+            return true;
```

Comment:
It's a finer point and not a functional concern but I like to see methods with imperative names return void.